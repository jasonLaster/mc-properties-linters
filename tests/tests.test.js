const fs = require("fs");
const path = require("path");
const { parse } = require("../src/parse");

const { hasLabel, checkKey, testGroup, testFile } = require("../src/tests");

function getSource(name) {
  return fs.readFileSync(path.join(__dirname, `./fixtures/${name}`), "utf8");
}

describe("tests", () => {
  describe("hasLabel", () => {
    it("good", () => {
      const text = getSource("simple.txt");
      const groups = parse(text);

      expect(hasLabel(groups[0])).toBe(true);
    });

    it("bad", () => {
      const text = getSource("bad.txt");
      const groups = parse(text);

      expect(hasLabel(groups[0])).toBe(false);
    });
  });

  describe("checkKey", () => {
    it("good", () => {
      const text = getSource("simple.txt");
      const groups = parse(text);

      expect(checkKey(groups[0])).toBe(true);
    });

    it("bad", () => {
      const text = getSource("bad.txt");
      const groups = parse(text);

      expect(checkKey(groups[1])).toBe(false);
      expect(checkKey(groups[2])).toBe(false);
    });
  });

  describe("testGroup", () => {
    it("good", () => {
      const text = getSource("simple.txt");
      const groups = parse(text);

      expect(testGroup(groups[0])).toBe(false);
      expect(testGroup(groups[1])).toBe(false);
    });

    fit("bad", () => {
      const text = getSource("bad.txt");
      const groups = parse(text);
      expect(testGroup(groups[0])).toMatchSnapshot();
      expect(testGroup(groups[1])).toMatchSnapshot();
      expect(testGroup(groups[2])).toMatchSnapshot();
    });
  });

  describe("testFile", () => {
    it("good", () => {
      const text = getSource("simple.txt");
      const groups = parse(text);

      expect(testFile(groups)).toBe(true);
    });

    it("bad", () => {
      const text = getSource("bad.txt");
      const groups = parse(text);
      expect(testFile(groups)).toMatchSnapshot();
    });
  });
});
