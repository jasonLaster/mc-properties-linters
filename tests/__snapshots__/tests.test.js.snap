// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests testFile bad 1`] = `
"Oops, missing a label.
# LOCALIZATION NOTE copySource): This is the text that appears in the

Oops, an incorrect key.
 # LOCALIZATION NOTE (copySource): This is the text that appears in the

Oops, an incorrect key.
 # LOCALIZATION NOTE (copySource): This is the text that appears in the"
`;

exports[`tests testGroup bad 1`] = `
"Oops, missing a label.
# LOCALIZATION NOTE copySource): This is the text that appears in the"
`;

exports[`tests testGroup bad 2`] = `
"Oops, an incorrect key.
 # LOCALIZATION NOTE (copySource): This is the text that appears in the"
`;

exports[`tests testGroup bad 3`] = `
"Oops, an incorrect key.
 # LOCALIZATION NOTE (copySource): This is the text that appears in the"
`;
